File: app/src/test/java/com/weiqi/sgf/SgfImportTest.kt
package com.weiqi.sgf

import com.weiqi.engine.CellState
import com.weiqi.engine.Point
import org.junit.Assert.assertEquals
import org.junit.Test

class SgfImportTest {
    @Test fun importsHeaderAndMoves() {
        val sgf = "(;GM[1]FF[4]SZ[9]KM[7.5]HA[0];B[dc];W[ed];B[];W[ef])"
        val s = SgfImport.import(sgf)
        assertEquals(9, s.config.boardSize)
        assertEquals(7.5, s.config.komi, 1e-9)
        assertEquals(CellState.BLACK, s.board.get(Point(2, 3)))    // dc -> col d (3), row c (2)
        assertEquals(CellState.WHITE, s.board.get(Point(3, 4)))
        assertEquals(CellState.WHITE, s.board.get(Point(5, 4)))
    }
}

File: test/sgf/sgf_import_test.dart
import 'package:flutter_test/flutter_test.dart';
import 'package:weiqi/src/domain/models.dart';
import 'package:weiqi/src/sgf/sgf_import.dart';

void main() {
  test('imports header and moves', () {
    const sgf = '(;GM[1]FF[4]SZ[9]KM[7.5]HA[0];B[dc];W[ed];B[];W[ef])';
    final s = SgfImport.import(sgf);
    expect(s.config.boardSize, 9);
    expect(s.config.komi, closeTo(7.5, 1e-9));
    expect(s.board.cellAt(const Point(2, 3)), CellState.black);
    expect(s.board.cellAt(const Point(3, 4)), CellState.white);
    expect(s.board.cellAt(const Point(5, 4)), CellState.white);
  });
}
